edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Interfaces/IMetricsFileStorage.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    /// <summary>
+    /// Stores metrics for later upload.
+    /// </summary>
+    public interface IMetricsFileStorage
+    {
+        /// <summary>
+        /// Adds the string to its storage. The current date will be recorded with it.
+        /// </summary>
+        /// <param name="data">data to store.</param>
+        void AddScrapeResult(string data);
+
+        /// <summary>
+        /// Retrieves all stored data.
+        /// </summary>
+        /// <returns>Key is date stored and value is a function that retuens the stored data.</returns>
+        IDictionary<DateTime, Func<string>> GetData();
```

Comment:
Since the volume of data may be huge, the return type should use stream or IEnumerable instead.

Reply:
Also should it be Async method?

Reply:
The volume shouldn't ever be to large (it is stored as a string, and it's only a single scrape per file) and it has to be parsed as a string by newtonsoft, so I don;t think it makes sense to make this a stream.